src/Sylius/Component/Core/Distributor/ProportionalIntegerDistributor.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Distributor;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class ProportionalIntegerDistributor implements ProportionalIntegerDistributorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function distribute(array $integers, $amount)
+    {
+        $total = array_sum($integers);
+        $distributedAmounts = [];
+
+        foreach ($integers as $element) {
```

Comment:
naming of this variable is a little bit confusing to me, but I don't have better idea for them